Makefile

```diff
@@ -72,3 +76,59 @@ update-kubernetes-proto: proto-tools
 		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/runtime \
 		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/schema/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/schema \
 		--go-plugin_opt=Mk8s.io/apimachinery/pkg/util/intstr/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/instr;
+	@$(MAKE) format
+
+golangci_lint_path := $(shell go env GOPATH)/bin/golangci-lint
+
+$(golangci_lint_path):
+	@go install $(golangci_lint)
+
+golangci_lint_goarch ?= $(shell go env GOARCH)
+
+.PHONY: lint
+lint: $(golangci_lint_path)
+	@for f in $(all_mods); do \
+        (cd $$(dirname $$f); GOARCH=$(golangci_lint_goarch) CGO_ENABLED=0 $(golangci_lint_path) run --timeout 5m); \
+	done
+
+.PHONY: format
+format:
+	@go run $(gofumpt) -l -w .
+	@go run $(gosimports) -local sigs.k8s.io/kube-scheduler-wasm-extension/ -w $(shell find . -name '*.go' -type f)
+
+# all_mods are the go modules except examples. Even though the guest modules
+# will be ultimately compiled with TinyGo, compiling with Go should also work.
+all_examples :=  $(wildcard ./examples/*/go.mod)
+all_mods     := ./internal/e2e/go.mod ./scheduler/go.mod ./guest/go.mod ./kubernetes/proto/go.mod $(all_examples)
+
+.PHONY: tidy
+tidy:
+	@for f in $(all_mods); do \
+        (cd $$(dirname $$f); go mod tidy); \
+	done
+
+.PHONY: build
+build:
+	@# We filter out the examples main packages, as nottinygo cannot compiled \
+     # to a normal platform executable.
+	@for f in $(filter-out $(all_examples), $(all_mods)); do \
+        (cd $$(dirname $$f); go build ./...); \
+	done
+
+.PHONY: test-scheduler
+test-scheduler:
+	@for d in ./scheduler ./internal/e2e; do \
+        (cd $$d; go test ./...); \
+	done
+
+.PHONY: check  # Pre-flight check for pull requests
+check:
+	@$(MAKE) tidy
+	@$(MAKE) build
+	@$(MAKE) lint golangci_lint_goarch=arm64
```

Comment:
I guess there isn't much of a chance of having amd/arm-specific code in this repo like wazero, so does it make sense to just run lint once with default settings, using `go run`? I do sometimes have issues in the wazero repo because of version drift of the installed golangci binary, but don't know of any alternative there because of the arch-specific code. If it's not an issue in this repo, `go run` may be better

Reply:
good point only once makes sense

Reply:
A lot better, thanks for calling attention to it!